src/rust/lib/graph-editor/src/component/node/connection.rs

```diff
@@ -0,0 +1,384 @@
+//! This module defines the shapes and logic required for drawing node connections.
+use crate::prelude::*;
+
+use crate::frp;
+use crate::component::node::port::{WeakPort, OutputPortView, InputPortView, OutputPort, InputPort};
+
+use enso_prelude::StrongRef;
+use ensogl::data::color::*;
+use ensogl::display::Attribute;
+use ensogl::display::Buffer;
+use ensogl::display::Scene;
+use ensogl::display::layout::alignment;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::ShapeRegistry;
+use ensogl::display::shape::*;
+use ensogl::display::shape::primitive::def::class::ShapeOps;
+use ensogl::display::symbol::geometry::Sprite;
+use ensogl::display;
+use ensogl::gui::component::ShapeViewDefinition;
+use ensogl::gui::component;
+use ensogl::math::geometry::triangle::Triangle;
+
+
+
+// ========================
+// === Connection Shape ===
+// ========================
+
+/// Defines a shape based ona a starting point and an end point. Supports a glow effect.
+///
+/// At the moment this is implemented as a simple line that is rotated and translated to match
+/// up with the start and end point.
+/// FIXME make this nicer. For example, use splines.
+mod shape {
+    use super::*;
+
+    ensogl::define_shape_system! {
+         (start:Vector2<f32>, end:Vector2<f32>, thickness:f32, glow:f32) {
+
+            let height     = Var::<f32>::from("input_start.y - input_end.y");
+            let width      = Var::<f32>::from("input_start.x - input_end.x");
+            let is_reverse = Var::<f32>::from("((((-sign(input_start.y-input_end.y)))))");
+
+            let angle_inner = Var::<f32>::from(90_f32.to_radians());
+            let triangle    = Triangle::<Var<f32>>::from_sides_and_angle(height,width,angle_inner);
+            let angle_a     = Var::<Angle<Radians>>::from(is_reverse * triangle.angle_a().clone());
```

Comment:
why connection uses triangles and glows?

Reply:
No more triangles.
Glow is there as a hover effect.